src/pytorch_lightning/trainer/trainer.py

```diff
@@ -225,20 +221,14 @@ def __init__(
                 that don't support deterministic mode (requires PyTorch 1.11+). If not set, defaults to ``False``.
                 Default: ``None``.
 
-            devices: Will be mapped to either `gpus`, `tpu_cores`, `num_processes` or `ipus`,
-                based on the accelerator type.
+            devices: The devices to use. Can be set to a positive number (int or str), a sequence of device indices
+                (list or str), the value ``-1`` to indicate all available devices should be used, or ``"auto"`` for
+                automatic selection based on the chosen accelerator. Default: ``"auto"``.
```

Comment:
Isn't the default 1?

Reply:
No, it is auto:

```py
    def _set_devices_flag_if_auto_passed(self) -> None:
        if self._devices_flag == "auto" or self._devices_flag is None:
            self._devices_flag = self.accelerator.auto_device_count()
```

Reply:
We can consider making a follow up that sets this explicitly in the constructor as the default value rather than None

Reply:
It says:

```py
ERROR: not found: /__w/2/s/tests/tests_pytorch/models/test_tpu.py::test_tpu_devices_with_argparse[--accelerator=tpu
(no name '/__w/2/s/tests/tests_pytorch/models/test_tpu.py::test_tpu_devices_with_argparse[--accelerator=tpu' in any of [<DoctestModule tests/tests_pytorch/models/test_tpu.py>, <Module tests/tests_pytorch/models/test_tpu.py>])
```

Reply:
That would be good.